src/components/clusterexplorer/node.ts

```diff
@@ -81,7 +85,29 @@ export type ClusterExplorerNode =
     ClusterExplorerHelmHistoryNode |
     ClusterExplorerCustomNode;
 
+export type ClusterExplorerNodeV2 =
+    ClusterExplorerMessageNode |
+    ClusterExplorerClusterNode |
+    ClusterExplorerContextNode |
+    ClusterExplorerResourceFolderNode |
+    ClusterExplorerGroupingFolderNode |
+    ClusterExplorerResourceNode |
+    ClusterExplorerConfigurationValueNode |
+    ClusterExplorerHelmReleaseNode |
+    ClusterExplorerHelmHistoryNode |
+    ClusterExplorerCustomNode;
+
 export class ClusterExplorerNodeImpl {
     readonly nodeCategory = KUBERNETES_EXPLORER_NODE_CATEGORY;
+    customizedTreeItem?: vscode.TreeItem;
     constructor(readonly nodeType: KubernetesExplorerNodeType) {}
+    getTreeItem(): vscode.TreeItem | Thenable<vscode.TreeItem> {
+        if (this.customizedTreeItem) {
+            return this.customizedTreeItem;
+        }
+        return this.getTreeItemInternal();
+    }
+    getTreeItemInternal(): vscode.TreeItem | Thenable<vscode.TreeItem> {
```

Comment:
How about `getBaseTreeItem` rather than `internal`?  All of this is pretty internal but the significance of this is that it's the initial object on top of which any customisations are applied.